### ISSUE TITLE
Many C4275 warning compiling on Windows

### ISSUE THREAD

**Issue by AWhetter** [openexr/openexr#182]
When compiling the shared libraries of IlmBase on Windows, there are many print outs of the warning

```
IlmBase\Iex\IexBaseExc.h(143) : warning C4275: non dll-interface class 'Iex::BaseExc' used as base for dll-interface class 'Iex::ArgExc'
        IlmBase\Iex\IexBaseExc.h(60) : see declaration of 'Iex::BaseExc'
        IlmBase\Iex\IexBaseExc.h(143) : see declaration of 'Iex::ArgExc'
```

This also occurs for every other exception in IexBaseExc.h at inherits from Iex::BaseExc.

When compiling external projects that use Iex, the error shows there as well.

**Comment by meshula** [comment]
The problem stems from the fact that Iex::BaseExc is subclassed from std::string which does not have a DLL adornment. Although it is easy to refactor Iex::BaseEXc so that it does not derive from string, that is unfortunately a breaking change for some applications.

**Comment by cary-ilm** [comment]
Looking into the OpenEXR issue backlog.  This was addressed in v2.3.0, Iex::BaseExc is no longer derived from std::string.
